filer/models/filemodels.py

```diff
@@ -469,8 +474,8 @@ def restore(self):
             self.folder = filer.models.Folder.objects.get(id=self.folder_id)
 
         old_location, new_location = self.file.name, None
-        destination = self._generate_valid_location_for_restore()
-
+        self._generate_valid_name_for_restore()
+        destination = self.file.field.upload_to(self, self.upload_to_name)
```

Comment:
I is a good idea to keep the details of generating the destination in the previous method. Is there a reason you decided to split the location name generation?
Also, for reasons of code readability the generate name function should return something instead of None. 
It would be easier to understand the return value instead of just mutating the object's internal state.